File: BackEnd/Application/UseCases/Comments/UpdateComment/UpdateCommentCommandHandler.cs
using Domain.Aggregates;
using Infrastructure.Repository;
using Infrastructure.UnitOfWork;
using MediatR;

namespace Application.UseCases.Comments.UpdateComment;

public class UpdateCommentCommandHandler(IRepository<Comment, Guid> commentRepository, IUnitOfWork unitOfWork) : IRequestHandler<UpdateCommentCommand, bool>
{
    public async Task<bool> Handle(UpdateCommentCommand request, CancellationToken cancellationToken)
    {
        var comment = await commentRepository.GetByIdAsync(request.Id, cancellationToken);
        if (comment == null) return false;
        comment.Update(request.Content);
        await commentRepository.UpdateAsync(comment);
        await unitOfWork.SaveChangesAsync(cancellationToken);
        return true;

    }
}

File: flutter/lib/application/use_cases/comments/update_comment/update_comment_command_handler.dart
import 'package:enterprise_management/application/use_cases/comments/update_comment/update_comment_command.dart';
import 'package:enterprise_management/infrastructure/repositories/comment_repository.dart';
import 'package:enterprise_management/shared_kernel/cqrs/command_handler.dart';
import 'package:enterprise_management/shared_kernel/result/result.dart';

class UpdateCommentCommandHandler extends ICommandHandler<UpdateCommentCommand, bool> {
  const UpdateCommentCommandHandler({required this._commentRepository});
  final ICommentRepository _commentRepository;

  @override
  Future<Result<bool>> handle(UpdateCommentCommand command) async {
    final res = await _commentRepository.updateComment(id: command.id, content: command.content);
    return Result.success(res);
  }

}